microsoft/vscode
# Add support for up to 24 bracket pair colorization colors

This PR attempts to fix #139777 @hediet

I'm also willing to take a look at the other solution proposed by @avicularian in https://github.com/microsoft/vscode/issues/139777#issuecomment-1191864893

Description:
- implements approach proposed by @hediet and @avicularian
- adds `editorBracketHighlightingForeground7..24`
- adds `editorBracketPairGuideBackground7..24`
- adds `editorBracketPairGuideActiveBackground7..24`
- note: I'm not sure how to proceed with the `% 30` modulo in `colorizedBracketPairsDecorationProvider.ts` L125 - see L126, especially since 30 is not the LCM of 1..6, which would be 60

How to test:

- apply provided `settings.json`
- open provided `test.json` and run `Format Document` command, or use any other applicable file with bracket pair colorization usage
- verify correct colorization (bracket colors, active bracket pair guide colors)
- in `settings.json`, switch ` "editor.guides.bracketPairs"` from `"active"` to `true` to verify inactive background pair guide colors
- verify correct colorization (bracket colors, inactive bracket pair guide colors)

`settings.json`
```
{
    "editor.bracketPairColorization.enabled": true,
    "editor.bracketPairColorization.independentColorPoolPerBracketType": false,
    "editor.guides.bracketPairs": "active",
    "workbench.colorCustomizations": {
        "editorBracketHighlight.unexpectedBracket.foreground": "#ffffff",
        "editorBracketHighlight.foreground1": "#ff0000",
        "editorBracketHighlight.foreground2": "#ff8000",
        "editorBracketHighlight.foreground3": "#ffbf00",
        "editorBracketHighlight.foreground4": "#ffff00",
        "editorBracketHighlight.foreground5": "#bfff00",
        "editorBracketHighlight.foreground6": "#00ff00",
        "editorBracketHighlight.foreground7": "#00ffbf",
        "editorBracketHighlight.foreground8": "#00ffff",
        "editorBracketHighlight.foreground9": "#00bfff",
        "editorBracketHighlight.foreground10": "#0080ff",
        "editorBracketHighlight.foreground11": "#0066ff",
        "editorBracketHighlight.foreground12": "#0000ff",
        "editorBracketHighlight.foreground13": "#8000ff",
        "editorBracketHighlight.foreground14": "#bf00ff",
        "editorBracketHighlight.foreground15": "#ff00ff",
        "editorBracketHighlight.foreground16": "#ff00bf",
        "editorBracketHighlight.foreground17": "#ff0080",
        "editorBracketPairGuide.background1": "#ff0000",
        "editorBracketPairGuide.background2": "#ff8000",
        "editorBracketPairGuide.background3": "#ffbf00",
        "editorBracketPairGuide.background4": "#ffff00",
        "editorBracketPairGuide.background5": "#bfff00",
        "editorBracketPairGuide.background6": "#00ff00",
        "editorBracketPairGuide.background7": "#00ffbf",
        "editorBracketPairGuide.background8": "#00ffff",
        "editorBracketPairGuide.background9": "#00bfff",
        "editorBracketPairGuide.background10": "#0080ff",
        "editorBracketPairGuide.background11": "#0066ff",
        "editorBracketPairGuide.background12": "#0000ff",
        "editorBracketPairGuide.background13": "#8000ff",
        "editorBracketPairGuide.background14": "#bf00ff",
        "editorBracketPairGuide.background15": "#ff00ff",
        "editorBracketPairGuide.background16": "#ff00bf",
        "editorBracketPairGuide.background17": "#ff0080",
        "editorBracketPairGuide.activeBackground1": "#ff0000",
        "editorBracketPairGuide.activeBackground2": "#ff8000",
        "editorBracketPairGuide.activeBackground3": "#ffbf00",
        "editorBracketPairGuide.activeBackground4": "#ffff00",
        "editorBracketPairGuide.activeBackground5": "#bfff00",
        "editorBracketPairGuide.activeBackground6": "#00ff00",
        "editorBracketPairGuide.activeBackground7": "#00ffbf",
        "editorBracketPairGuide.activeBackground8": "#00ffff",
        "editorBracketPairGuide.activeBackground9": "#00bfff",
        "editorBracketPairGuide.activeBackground10": "#0080ff",
        "editorBracketPairGuide.activeBackground11": "#0066ff",
        "editorBracketPairGuide.activeBackground12": "#0000ff",
        "editorBracketPairGuide.activeBackground13": "#8000ff",
        "editorBracketPairGuide.activeBackground14": "#bf00ff",
        "editorBracketPairGuide.activeBackground15": "#ff00ff",
        "editorBracketPairGuide.activeBackground16": "#ff00bf",
        "editorBracketPairGuide.activeBackground17": "#ff0080",
    }
}
```

`test.json` (not valid json - irrelevant for testing)
```
[[[[[[[[[[{{{{{{{{{{{{{{{{{{{{{{{{[[[[{}]]]]}}}}}}}}}}}}}}}}}}}}}}}}]]]]]]]]]]
```
